service/src/main/java/bio/terra/workspace/service/petserviceaccount/PetSaService.java

```diff
@@ -74,12 +77,13 @@ public Policy enablePetServiceAccountImpersonation(UUID workspaceId, UserWithPet
    *
    * @param workspaceId ID of the workspace to enable pet SA in
    * @param userAndPet The user and the pet SA the user is being granted permission to
+   * @param proxyGroupEmail User's proxy group
    * @param eTag GCP eTag which must match the pet SA's current policy. If null, this is ignored.
    * @return The new IAM policy on the user's pet service account, or empty if the eTag value
    *     provided is non-null and does not match current IAM policy on the pet SA.
    */
   public Optional<Policy> enablePetServiceAccountImpersonationWithEtag(
-      UUID workspaceId, UserWithPetSa userAndPet, @Nullable String eTag) {
+      UUID workspaceId, UserWithPetSa userAndPet, String proxyGroupEmail, @Nullable String eTag) {
```

Comment:
Do you still need a `UserWithPetSa` for this method if we're only getting the user email from it now? You could just take in the user email directly.

Reply:
On line 89, we get the pet email (to look up pet SA).

Reply:
my bad.